tests/core/test_graph.py

```diff
@@ -164,6 +164,20 @@ def test_graph_constructor_nodes_from_edges():
     assert g.node_types == {"abc"}
     assert sorted(g.nodes()) == [0, 1, 2, 4, 5]
 
```

Comment:
Could you add a comment describing why we're adding these new tests? As it is, I don't think it'll be at all obvious to someone reading this in future why these tests are here. E.g.

```suggestion

    # node inference shouldn't hide the real errors in an invalid 'edges' param
```

Reply:
yeah good point, they're super non obvious :+1: